h2/src/main/org/h2/jdbcx/JdbcConnectionPool.java

```diff
@@ -217,17 +226,14 @@ public Connection getConnection(String user, String password) {
     }
 
     private Connection getConnectionNow() throws SQLException {
-        if (isDisposed) {
+        if (isDisposed.get()) {
             throw new IllegalStateException("Connection pool has been disposed.");
         }
-        PooledConnection pc;
-        if (!recycledConnections.isEmpty()) {
-            pc = recycledConnections.remove(recycledConnections.size() - 1);
-        } else {
+        PooledConnection pc = recycledConnections.poll();
+        if (pc == null) {
             pc = dataSource.getPooledConnection();
         }
         Connection conn = pc.getConnection();
-        activeConnections++;
```

Comment:
you dropped the increment of activeConnections?

Reply:
activeConnections was increment at line 197, outside the call of function getConnectionNow